### ISSUE TITLE
Inconsistent set/get for cascading keys

### ISSUE THREAD

**Issue by omnidan** [ElektraInitiative/libelektra#401]
I have a `test.ini` file with the following contents mounted at `/test`:

```
[test]
hello = world
```

After running the following commands:

```
dan@vagrant-ubuntu-vivid-64:~/libelektra/build/bin$ ./kdb get /test/test/hello
world
dan@vagrant-ubuntu-vivid-64:~/libelektra/build/bin$ ./kdb set /test/test/hello earth
Using name user/test/test/hello
Create a new key user/test/test/hello with string earth
dan@vagrant-ubuntu-vivid-64:~/libelektra/build/bin$ ./kdb get /test/test/hello
earth
```

The key gets written to KDB, but not to the mounted file - this seems to be an issue with the resolver.

Using `system/test` as a mountpoint works fine.

**Comment by markus2330** [comment]
Be extra careful with cascading names, you might get something different than you write to. You can use `-v` to check.

**Comment by omnidan** [comment]
Sorry for the late response, I kinda overlooked this issue. :sweat_smile: Output with `-v`: 

```
dan@vagrant-ubuntu-vivid-64:~/libelektra/build/bin$ ./kdb get -v /test/test/hello
got 9 keys
searching spec/test/test/hello, found: <nothing>, options: KDB_O_CALLBACK
    searching proc/test/test/hello, found: <nothing>, options: 
    searching dir/test/test/hello, found: dir/test/test/hello, options: 
The resulting keyname is dir/test/test/hello
earth
```

**Comment by markus2330** [comment]
Ok, seems like you actually got the key `dir/test/test/hello` and not `user/test/test/hello` as you expected.

Writing to cascading keys is sometimes not intutive, I agree that it is not a good idea that a `get` afterwards should return what you have `set` before.

Maybe we should prohibit the writing to cascading keys at all, it causes more problems than it solves.

**Comment by markus2330** [comment]
Please reopen on any idea how to improve this. Documentation got some updates and `-v` tells quite good what `kdb get` does.

**Comment by markus2330** [comment]
Reopened: we should at least warn:

- if a `kdb set`is hidden by an already present key
- if `kdb set/rm` points to some other keys

**Comment by markus2330** [comment]
To be solved with #3742
